packages/devtools/lib/src/timeline/timeline_protocol.dart

```diff
@@ -326,6 +370,29 @@ class TimelineData {
       frame.addedToTimeline = true;
     }
   }
+
+  TimelineEventType _inferEventType(TraceEvent event) {
+    if (event.threadId == cpuThreadId) {
+      return TimelineEventType.cpu;
+    } else if (event.threadId == gpuThreadId) {
+      return TimelineEventType.gpu;
+    } else {
+      return TimelineEventType.unknown;
+    }
+  }
+
+  bool _shouldProcessTraceEvent(TraceEvent event) {
+    const List<String> phaseWhitelist = ['s', 'f', 'B', 'E', 'X'];
+    return phaseWhitelist.contains(event.phase) &&
+        // Do not process Garbage Collection events.
+        event.category != 'GC' &&
+        // Do not process MessageLoop::RunExpiredTasks events. These events can
+        // either a) start outside of our frame start time, b) parent irrelevant
+        // events, or c) parent multiple event flows - none of which we want.
+        event.name != 'MessageLoop::RunExpiredTasks' &&
+        // Only process events from the CPU or GPU thread.
+        (event.isGpuEvent || event.isCpuEvent);
```

Comment:
what events aren't gpu or cpu?

Reply:
If we received events from another thread, these would be neither cpu or gpu events. We want to make sure we don't process these.